### PR TITLE
Add Enqueued_Styles_Size_Check

### LINKED ISSUE #19: Create Enqueued_Styles_Size_Check
## Description

This check should work the same as [Enqueued_Scripts_Size_Check](https://github.com/10up/plugin-check/issues/17), adapted for styles instead of scripts.

## Acceptance Criteria

- Class `Enqueued_Styles_Size_Check` should be created and exists at `includes/Checker/Checks`
- Class `Enqueued_Styles_Size_Check` should extend the `Runtime_Check` abstract class which extends `Check` interface
- Property `$violation` code should be `warning`
- Method `run` should be implemented from the `Runtime_Check` abstract class if needed
- `Check_Result $check_result` should be used from the `run` method parameter to append the results

## Tests Coverage
- `Enqueued_Styles_Size_Check` class is expected to extend `Runtime_Check` class
- `Enqueued_Styles_Size_Check` class is expected to be instance of `Runtime_Check` class
- `run` method expected to append the audit results to `$check_result`

### PR DESCRIPTION
**Pull request by ernilambar**
Fixes https://github.com/WordPress/plugin-check/issues/19

**Note:**

- I have added tests to cover my change.
- All new and existing tests pass.

### CODE REVIEW

**Comment by davidperezgar** [comment]
I've changed the tag so the Performance Team can review it.

File: includes/Checker/Checks/Enqueued_Styles_Size_Check.php [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+		if ( $plugin_style_size > $this->threshold_size ) {
+			foreach ( $plugin_styles as $plugin_style ) {
+				$this->add_result_warning_for_file(
+					$result,
+					sprintf(
+						'This style has a size of %1$s which in combination with the other styles enqueued on %2$s exceeds the style size threshold of %3$s.',
+						size_format( $plugin_style['size'] ),
+						$url,
+						size_format( $this->threshold_size )
+					),
```
Comment (swissspidy) [inline includes/Checker/Checks/Enqueued_Styles_Size_Check.php]: I just realized `Enqueued_Scripts_Size_Check` has the same flaw, so would be great to make the string translatable there as well.

```suggestion
					sprintf(
						/* translators: 1: style file size. 2: tested URL. 3: threshold file size. */
						__( 'This style has a size of %1$s which in combination with the other styles enqueued on %2$s exceeds the style size threshold of %3$s.', 'plugin-check' )
						size_format( $plugin_style['size'] ),
						$url,
						size_format( $this->threshold_size )
					),
```

**Review by swissspidy — APPROVED** [review]
IDK why this wasn't picked up earlier, apologies!

Given that this adapts `Enqueued_Scripts_Size_Check` for styles, it looks pretty straightforward.

Just one suggestion regarding i18n.

**Comment by swissspidy** [comment]
Related:

* #74

Maybe we could add this here as well from the beginning?

File: includes/Checker/Checks/Enqueued_Styles_Size_Check.php [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+use WordPress\Plugin_Check\Checker\Preparations\Demo_Posts_Creation_Preparation;
+use WordPress\Plugin_Check\Checker\With_Shared_Preparations;
+use WordPress\Plugin_Check\Traits\Amend_Check_Result;
+use WordPress\Plugin_Check\Traits\Stable_Check;
+use WordPress\Plugin_Check\Traits\URL_Aware;
+
+/**
+ * Check for enqueued style sizes.
+ *
+ * @since n.e.x.t
```
Comment (mukeshpanchal27) [inline includes/Checker/Checks/Enqueued_Styles_Size_Check.php]: Add specific version number as we don't have shell script that change `n.e.x.t` to upcoming version when releasing.

May be `1.0.2` is next version

File: includes/Checker/Checks/Enqueued_Styles_Size_Check.php [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+			$plugin_style_size += $style_size;
+		}
+
+		if ( $plugin_style_size > $this->threshold_size ) {
+			foreach ( $plugin_styles as $plugin_style ) {
+				$this->add_result_warning_for_file(
+					$result,
+					sprintf(
+						/* translators: 1: style file size. 2: tested URL. 3: threshold file size. */
+						__( 'This style has a size of %1$s which in combination with the other styles enqueued on %2$s exceeds the style size threshold of %3$s.', 'plugin-check' )
```
Comment (mukeshpanchal27) [inline includes/Checker/Checks/Enqueued_Styles_Size_Check.php]:
```suggestion
						__( 'This style has a size of %1$s which in combination with the other styles enqueued on %2$s exceeds the style size threshold of %3$s.', 'plugin-check' ),
```

It shows error.

**Review by mukeshpanchal27 — CHANGES_REQUESTED** [review]

File: includes/Checker/Checks/Enqueued_Styles_Size_Check.php (hunk 3 above)
Comment (swissspidy) [reply includes/Checker/Checks/Enqueued_Styles_Size_Check.php]: Whoops my bad :)

**Comment by swissspidy** [comment]
In a separate PR we can look into creating an abstract class or so to reduce code repetition between this class and `Enqueued_Scripts_Size_Check`

**Review by westonruter — APPROVED** [review]

**Review by mukeshpanchal27 — APPROVED** [review]

**Pull request merged**